web/src/stream_list.js

```diff
@@ -39,16 +39,21 @@ export function update_count_in_dom($stream_li, count, stream_has_any_unread_men
     // count is there or not.
     const $subscription_block = $stream_li.find(".subscription_block");
 
-    ui_util.update_unread_count_in_dom($subscription_block, count);
+    ui_util.update_unread_count_in_dom($subscription_block, count.stream_count);
     ui_util.update_unread_mention_info_in_dom(
         $subscription_block,
         stream_has_any_unread_mention_messages,
     );
 
-    if (count === 0) {
+    if (count.stream_count === 0) {
         $subscription_block.removeClass("stream-with-count");
     } else {
         $subscription_block.addClass("stream-with-count");
+        if (count.has_unmute_unread) {
+            $subscription_block.addClass("has_unmute_unread");
+        } else {
+            $subscription_block.removeClass("has_unmute_unread");
+        }
```

Comment:
This function seemed hard to follow after these changes, especially with the `stream_count` variable name being quite poor, so I mostly rewrote it. 

Also the `count` variable name for an object passed in seems quite bad, so I changed tha too.

Finally, `has_unmute_unread` is a bad CSS class name for several reasons -- we prefer `-` separators in CSS class names, and it also is just had to read (and the `unmute` token is a weird choice instead of `unmuted`); changed to `has-unmuted-unreads`.